test/server/cards/01_SOR/leaders/CassianAndorDedicatedToTheRebellion.spec.ts

```diff
@@ -0,0 +1,130 @@
+describe('Cassian Andor, Dedicated to the Rebellion', function() {
+    integration(function(contextRef) {
+        describe('Cassian Andor\'s leader ability', function() {
+            it('should draw a card aftering dealing 3 damage to an enemy base with an attack', function() {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['daring-raid', 'daring-raid'],
+                        deck: ['k2so#cassians-counterpart', 'open-fire'],
+                        spaceArena: ['green-squadron-awing'],
+                        leader: { card: 'cassian-andor#dedicated-to-the-rebellion', deployed: false },
+                        resources: 4,
+                    },
+                    player2: {
+                        groundArena: ['wampa'],
+                        spaceArena: ['tieln-fighter'],
+                        base: { card: 'capital-city' }
+                    },
+                });
+                const { context } = contextRef;
+
+                // Expect this ability be available, but performs a no-op since 3 damage hasn't been dealt yet
+                expect(context.cassianAndor).toHaveAvailableActionWhenClickedBy(context.player1);
+                expect(context.cassianAndor.exhausted).toBeTrue();
+                expect(context.player1.exhaustedResourceCount).toBe(1);
+
+                context.cassianAndor.exhausted = false;
+                context.player2.passAction();
+
+                // Select the a-wing to deal 3 damage to base
+                context.player1.clickCard(context.greenSquadronAwing);
+                context.player1.clickCard(context.p2Base);
+                context.player2.passAction();
+
+                // Expect this ability have activated now - so it should draw the top card from the deck
+                expect(context.cassianAndor).toHaveAvailableActionWhenClickedBy(context.player1);
+                expect(context.cassianAndor.exhausted).toBeTrue();
+                expect(context.player1.exhaustedResourceCount).toBe(2);
+                expect(context.player1.hand).toContain(context.k2so);
+
+                context.moveToNextActionPhase();
+                context.player1.readyResources(4);
```

Comment:
Why is this needed? We're at the start of a new action phase so all resources should be ready

Reply:
vestigial, forgot to remove.  I thought resources weren't readying correctly, forgot that direct-fire was consuming resources (looked like it hadn't reset -- PEBKAC).  Cleaned up